src/genome.py

```diff
@@ -88,154 +97,260 @@ def __str__(self) -> str:
-        ...  # FIXME
-        return 0
+        """current_node length of the genome."""
+        return len(self.nucleotide)
 
     def __str__(self) -> str:
         """
         Return a string representation of the genome.
-
         Create a string that represents the genome. By nature, it will be
         linear, but imagine that the last character is immidiatetly followed
         by the first.
-
         The genome should start at position 0. Locations with no TE should be
         represented with the character '-', active TEs with 'A', and disabled
         TEs with 'x'.
         """
-        return "FIXME"
+        nucleotide = ""
+        for x in self.nucleotide:
+            if isinstance(x, int):
+                nucleotide += "A"
+            else: 
+                nucleotide += str(x)
+        return nucleotide
+
+class Link(Generic[T]):
+    """Doubly linked link.""" 
+
+    def __init__(self, val: T, p, n):
+        """Create a new link and link up prev and next."""
+        self.val = val
+        self.prev = p
+        self.next = n
+
+
+def insert_next(link: Link[T], val: T) -> None:
+    """Add a new link containing avl after link."""
+    new_link = Link(val, link, link.next)
+    new_link.prev.next = new_link
+    new_link.next.prev = new_link
+
+Ft = namedtuple("Ft", ["ft", "length"])
 
 
 class LinkedListGenome(Genome):
     """
     Representation of a genome.
-
     Implements the Genome interface using linked lists.
     """
+    head: Link[Ft]  
+    active = {}
+    counter = 0
 
     def __init__(self, n: int):
         """Create a new genome with length n."""
-        ...  # FIXME
+        self.head = Link(None, None, None)
+        self.head.next = self.head
+        self.head.prev = self.head
+        insert_next(self.head.prev, Ft(0, n))
 
     def insert_te(self, pos: int, length: int) -> int:
         """
         Insert a new transposable element.
-
         Insert a new transposable element at position pos and len
         nucleotide forward.
-
         If the TE collides with an existing TE, i.e. genome[pos]
         already contains TEs, then that TE should be disabled and
         removed from the set of active TEs.
-
         Returns a new ID for the transposable element.
         """
-        ...  # FIXME
-        return -1
+        for ft, end in self.iter_pos():
+            if end <= pos:
+                continue
+            else:
+                half_1 = ft.val.length -end + pos
+                half_2 = end - pos
+                self.insert_te_at_pos(length, ft, half_1, half_2)
+                return self.counter
+        return None
+
+    def insert_te_at_pos(self, length, ft, half_1, half_2):
+        if  ft.val.ft == 1:
+            self.disable_feature(ft)
+        insert_next(ft, Ft(1, length))
+        ft.val = Ft(ft.val.ft, half_1)
+        insert_next(ft.next, Ft(ft.val.ft, half_2))
+        self.counter += 1
+        self.active[self.counter] = ft.next
```

Comment:
One or both of the flanking regions can be empty here (i.e., `half_1` or `half_2` could be zero). Ignoring these edge cases will not cause any trouble, but if you end up with a lot of empty intervals it will slow down the code. Other than that, it is a perfect solution.